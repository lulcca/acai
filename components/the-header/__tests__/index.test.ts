import { beforeEach, describe, expect, it } from 'vitest';
import TheHeader from '@/components/the-header/index.vue';
import { mount } from '@vue/test-utils';

describe('Components > TheHeader', () => {
  let wrapper: ReturnType<typeof mount>;

  beforeEach(() => {
    wrapper = mount(TheHeader);
  });

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render title correctly', () => {
    expect(wrapper.find('h1').text()).toMatch(/Controle de açaí/i);
  });

  it('should render paragraph correctly', () => {
    expect(wrapper.find('p').text()).toMatch(/teste sua sorte e faça o atrasado pagar/i);
  });
});
