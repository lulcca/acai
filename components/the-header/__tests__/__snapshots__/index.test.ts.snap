// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Components > TheHeader > should match the snapshot 1`] = `
"<header class="gap-2 grid h-full place-items-center w-full">
  <h1 class="font-major text-8xl uppercase"> Controle de açaí </h1>
  <p> teste sua sorte e faça o atrasado pagar </p>
</header>"
`;
